### PR TITLE
Yui docs for plugin-loader.coffee

### PR DESCRIPTION
**Pull request by SteveMcArthur**

### CODE REVIEW

File: src/lib/plugin-loader.coffee [hunk 1]
```diff
@@ -15,50 +15,104 @@ safefs = require('safefs')
...
 
-	# Constructor
+	###*
+	# Constructor method
+	# @method constructor
+	# @param {Object} @docpad
+	# @param {String} @dirPath
+	# @param {Object} @BasePlugin
+	###
 	constructor: ({@docpad,@dirPath,@BasePlugin}) ->
```
Comment (balupton) [inline src/lib/plugin-loader.coffee]: Is that the correct convention, as what this code does is say we receive a opts argument, that contains `docpad`, `dirPath`, and `basePlugin` properties, which are then assigned to `this`

**Comment by SteveMcArthur** [comment]
OK - I couldn't quite understand what was going on here. But now that you explain it I get it. I haven't seen this sort of construction in the DocPad code before. Do you have a suggestion on how to document this? An opts argument with default values?

**Comment by SteveMcArthur** [comment]
Or are these 3 values required as the MUST be assigned to ```this```

**Comment by balupton** [comment]
> Or are these 3 values required as they MUST be assigned to this

All required. Plugin Loader is used just by DocPad Core. So DocPad Core instantiates the Plugin Loader class with those variables.

**Comment by SteveMcArthur** [comment]
mm - I get you. Still unsure how to document it. I think its three separate parameters all of which are required. Yuidocs doesn't (I think) have a concept of an object with required properties. I can leave as is but remove the "@" symbol from the parameter name. I think the "@" symbol just confuses things.

**Pull request merged**